src/lib/Strategy/Storage/GroupItemStrategyInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClient\Strategy\Storage;
+
+use Ibexa\Contracts\Personalization\Criteria\CriteriaInterface;
+use Ibexa\Contracts\Personalization\Storage\DataSourceInterface;
+use Ibexa\Contracts\Personalization\Value\ItemGroupListInterface;
+
+interface GroupItemStrategyInterface
+{
+    public function getGroupList(DataSourceInterface $source, CriteriaInterface $criteria): ItemGroupListInterface;
+
+    public static function getIndex(): string;
```

Comment:
If it's internal class then it's ok to make it a part of the interface, but remember that this means that you cannot create two instances (services) of the same class with different grouping indexes.

Ideally you'd want to have a note about the DI feature (that reads this static method) for autoconfiguration in phpdoc instead.